filebeat/prospector/plugin.go

```diff
@@ -0,0 +1,38 @@
+package prospector
+
+import (
+	"errors"
+
+	"github.com/elastic/beats/libbeat/plugin"
+)
+
+type prospectorPlugin struct {
```

Comment:
Thinking if we should call this just `plugin`  as it's already in the prospector package.

Reply:
I was thinking about it to. However, to stay consistent with other with existing plugin.go files I decided to name it `prospectorPlugin`. As other files these `struct`s are called `outputPlugin`, `modulePlugin`, etc.

Reply:
SGTM. Lets keep it in line for now.